CHANGELOG.md

```diff
@@ -1,10 +1,64 @@
-# 0.7.0 alpha.1
-
-Currently the changelog for v0.7.0 is incomplete, this tracked in issue #1217.
+# 0.7.0
+
+Version 0.7 of Mio contains various major changes compared to version 0.6.
+Overall a large number of API changes have been made to reduce the complexity of
+the implementation and remove overhead where possible.
+
+Also the upgrade guide in docs/upgrade_guide.md.
+
+## Added
+
+* New `Interest` structure that replaces `Ready` in registering event sources.
```

Comment:
`New` is a little redundant here as the section is titled `Added`. Maybe change this to: `'Interest' structure that ..`